File: algorithms/searching/binary search/binarySearch.js
const fs = require("fs");
const data = fs
  .readFileSync("values.txt", "utf8")
  .split(",")
  .map((x) => Number(x));

function binarySearch(values, target) {
  let low = 0;
  let mid = Math.floor(values.length / 2);
  let high = values.length - 1;

  while (low <= high) {
    mid = Math.floor((low + high) / 2);
    const guess = values[mid];

    if (guess === target) {
      console.log(`Found ${guess} at index ${mid}!`);
      return guess;
    } else if (guess < target) {
      low = mid + 1;
    } else if (guess > target) {
      high = mid - 1;
    }
  }
  console.log("Value not found!");
  return false;
}

binarySearch(data, 1);
binarySearch(data, 99);
binarySearch(data, 1110);

File: algorithms/searching/binary search/binary_search.py
data = open("values.txt", "r")
values = [int(v) for v in data.read().split(",")]

def binary_search(values, target):
    low = 0;
    mid = len(values)//2
    high = len(values)-1

    while low <= high:
        mid = (low + high)//2
        guess = values[mid]

        if guess == target:
            print(f"Found {target} at index {mid}!")
            return guess
        elif guess < target:
            low = mid + 1
        elif guess > target:
            high = mid - 1
    print("Value not found!")
    return None


binary_search(values, 1)
binary_search(values, 99)
binary_search(values, 1110)